readthedocs/builds/tasks.py

```diff
@@ -86,19 +86,20 @@ def route_for_task(self, task, args, kwargs, **__):
         if version.type == EXTERNAL:
             last_build_for_default_version = (
                 project.builds
-                .filter(version__slug=project.get_default_version())
+                .filter(version__slug=project.get_default_version(), builder__isnull=False)
```

Comment:
We are filtering by `success=True` in the queryset below as well.

Reply:
Yea, this is explicit about what we want tho. There could be a case where success=True, but builder=None for some reason.

Reply:
Maybe both filters? I'm not familiar with the algorithm we are using to decide the queue here. But I think we want to use the queue that was from a successful build instead of one that failed.

Reply:
Aye, true. @humitos have thoughts?

Reply:
I want to use the exact queue that was used by the default version in a previous build (even if it fail). The idea is to reproduce the exact behavior on the PR than on the Version once it's get merged.

So, if it fails on the PR it should fail on the Version and vice-versa.

Reply:
Cool - I'll add a comment and ship it.